src/com/skillsup/dao/UserInFileDaoImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.skillsup.dao;
+
+import com.skillsup.model.Gender;
+import com.skillsup.model.User;
+import com.skillsup.model.UserRole;
+
+import java.io.*;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+
+public class UserInFileDaoImpl implements UserDAO {
+    private static final String PATH_USER_DB_FILE = "E:\\onlineShop\\src\\com\\skillsup\\resources\\userDB.txt";
+    private static final User ADMIN = new User("admin", "admin", UserRole.ADMIN);
+
+    @Override
+    public void saveUser(User user) {
```

Comment:
Creating new FIleWriter on each user save is very reliable but very slow (in terms of performance) approach, its better to reuse ffileWriter in this class

Reply:
I would like to clarify how to put it into practice